packages/core/src/components/EditableTypography/EditableTypography.tsx

```diff
@@ -215,13 +212,13 @@ const EditableTypography: VibeComponent<EditableTypographyProps, HTMLElement> =
       }
     }, [autoSelectTextOnEditMode, isEditing]);
 
-    useEffect(() => {
+    useLayoutEffect(() => {
       if (!typographyRef.current) {
         return;
       }
 
       const { width } = typographyRef.current.getBoundingClientRect();
-      setInputWidth(width);
+      inputRef?.current?.style.setProperty("--input-width", `${width}px`);
```

Comment:
it doesn't trigger re-render, is it cool?

Reply:
or it's actually the fix itself - the fact we're not re-rendering?

Reply:
yep, we don't need a re-render but a recalc/style, so now it doesn't interfer with react's renders when the state changes